### PR TITLE
Bump Swift to Version 4.0.2

### PR DESCRIPTION
**Pull request by zisko**
Bump Swift Version to 4.0.2

rdar://34888471

### CODE REVIEW

File: lib/Basic/Version.cpp [hunk 1]
```diff
@@ -314,7 +314,7 @@ Optional<Version> Version::getEffectiveLanguageVersion() const {
   // a version newer than any released 4.x compiler, and the
   // "5" case should start returning getCurrentLanguageVersion. We should
   // also check for the presence of SWIFT_VERSION_PATCHLEVEL, and if that's
-  // set apply it to the "3" case, so that Swift 4.0.1 will automatically
+  // set apply it to the "3" case, so that Swift 4 will automatically
```
Comment (zisko) [inline lib/Basic/Version.cpp]: @jrose-apple should this read 3?

Comment (jrose-apple) [reply lib/Basic/Version.cpp]: The comment is correct as originally written. It's saying that under "-swift-version 3" patch levels will propagate onto the compatibility version. 4.0.1 was just an example; it's not a reference to the current version.

**Comment by zisko** [comment]
@swift test and merge

**Comment by zisko** [comment]
@swift please test and merge

**Comment by zisko** [comment]
@swift-ci please test and merge

**Comment by zisko** [comment]
@swift-ci test

**Comment by zisko** [comment]
@swift-ci test and merge

**Comment by shahmishal** [comment]
@swift-ci test and merge

**Review by Casillazz — COMMENTED** [review]
Bueno

**Pull request merged**